superset/utils.py

```diff
@@ -841,3 +842,31 @@ def ensure_path_exists(path):
     except OSError as exc:
         if not (os.path.isdir(path) and exc.errno == errno.EEXIST):
             raise
+
+
+def get_query_without_limit(sql):
+    return re.sub(r"""
+                (?ix)        # case insensitive, verbose
+                \s+          # whitespace
+                LIMIT\s+\d+  # LIMIT $ROWS
+                ;?           # optional semi-colon
+                (\s|;)*$     # remove trailing spaces tabs or semicolons
+                """, '', sql)
+
+
+def get_limit_from_sql(sql):
+    # returns the limit of the quest or None if it has none
+    sql_without_limit = get_query_without_limit(sql)
+    if sql_without_limit != sql:
+        limit_clause = sql.partition(sql_without_limit)[2]
+        limit_match = re.search('limit\s(\d+)', limit_clause, re.IGNORECASE)
+        if limit_match:
+            specified_limit = limit_match.group(1)
+            try:
```

Comment:
This try/except is unnecessary as the regex explicitly matched a an integer. If the value was a non-integer `limit_match` would be `None`.

Reply:
True. Taken out completely.